### PR TITLE
[RHICOMPL-944] Configurable Kafka Producer Topics

### PR DESCRIPTION
**Pull request by vkrizan**
Refactored settings for Kafka topics configuration to section of consumers/subscriptions and producers. Made the consumers to use their respective settings.

### CODE REVIEW

**Review by akofink — COMMENTED** [review]
@vkrizan Thanks for thinking about this and refactoring the consumers as well; however, I don't think we should change them, since it requires updates to all of our deployment templates. In general, we should try to make the least invasive changes possible to maintain as much stability as possible. Perhaps @Victoremepunto or @bastilian also have some thoughts whether we should refactor these settings or not.

**Comment by vkrizan** [comment]
I haven't found any configuration overriding the default topics.

File: config/settings.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 app_name: 'compliance'
 path_prefix: '/api'
 old_path_prefix: '/r/insights/platform'
-platform_kafka_topic: 'platform.upload.compliance'
-platform_kafka_inventory_topic: 'platform.inventory.events'
-platform_kafka_validation_topic: 'platform.upload.validation'
+kafka_subscription_topics:
```
Comment (akofink) [inline config/settings.yml:4]: Should it be:
```suggestion
kafka_consumer_topics:
```
to match the `kafka_producer_topics`? That seems to make more sense to me when thinking of consumers and producers.

**Review by akofink — APPROVED** [review]
Ah, nice! I didn't realize we haven't overridden these settings. In that case, I think we're good! The new settings organization makes a lot of sense to me. :+1: I just had one nitpick on the naming of the consumer settings.

File: config/settings.yml (hunk 1 above)
Comment (vkrizan) [reply config/settings.yml]: Changed.

**Comment by Victoremepunto** [comment]
> @vkrizan Thanks for thinking about this and refactoring the consumers as well; however, I don't think we should change them, since it requires updates to all of our deployment templates. In general, we should try to make the least invasive changes possible to maintain as much stability as possible. Perhaps @Victoremepunto or @bastilian also have some thoughts whether we should refactor these settings or not.

@akofink  Yeh @vkrizan  is right, I didn't see anywhere in E2E where these values are replaced. we usually export a variable to the environment to overwrite those, I don't know how would this affect to that feature (now that we have 1 depth level), I guess we could use `SETTINGS__KAFKA_CONSUMER_TOPICS.INVENTORY_EVENTS` for that ? not sure how the config framework works for nested keys, to be honest :)

**Comment by akofink** [comment]
@Victoremepunto I think for more complex settings like this, we can use a ConfigMap, if needed, rather than env variables.

**Comment by vkrizan** [comment]
@Victoremepunto Each level from settings is divided by `__` (double underscore). So to set one topic, `SETTINGS__KAFKA_CONSUMER_TOPICS__INVENTORY_EVENTS` can be used.

**Pull request merged**